Source/DafnyCore/AST/Cloner.cs

```diff
@@ -815,7 +815,7 @@ class ScopeCloner : DeepModuleSignatureCloner {
       if (cf != null && cf.Rhs != null && !RevealedInScope(f)) {
         // We erase the RHS value. While we do that, we must also make sure the declaration does have a type, so instead of
         // cloning cf.Type, we assume "f" has been resolved and clone cf.Type.NormalizeExpandKeepConstraints().
-        return new ConstantField(Tok(cf.tok), cf.Name, null, cf.IsStatic, cf.IsGhost, cf.IsOpaque, CloneType(cf.Type.NormalizeExpandKeepConstraints()), CloneAttributes(cf.Attributes));
+        return new ConstantField(Tok(cf.tok), cf.Name, null, cf.HasStaticKeyword, cf.IsGhost, cf.IsOpaque, CloneType(cf.Type.NormalizeExpandKeepConstraints()), CloneAttributes(cf.Attributes));
```

Comment:
Is there a test that could catch this? I'm worrying that these two fields are too similar.

Reply:
I have not thought of or found one. IsStatic is computed as HasStaticKeyword or "is static by default".
So a problem would come when a const  declaration was not declared static, but is static by default, and
then the AST is cloned, and the HasStaticKeyword filed is used for something directly.
(This is a reason I'm not fond of C#'s computed properties)

Reply:
Then please approve it. But if you can tell me under what conditions such a cloine is made and for what purpose, I'll keep thinking about a test case.